src/pwncore/routes/leaderboard.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+from fastapi import APIRouter
+
+from pwncore.models import SolvedProblem, Team
+
+# Metadata at the top for instant accessibility
+metadata = {"name": "leaderboard", "description": "Operations on the leaderboard"}
+
+router = APIRouter(prefix="/leaderboard", tags=["leaderboard"])
+
+
+@router.get("")
+async def fetch_leaderboard():
+    teams = await Team.all()
+    leaderboard = {}
+    for team in teams:
+        problems_solved_by_team = await SolvedProblem.filter(
+            team__id=team.id
+        ).values_list("problem__points", flat=True)
+        # Tortoise's return type is List[Tuples] even after flat=True makes it List[]
+        # Hence ignore:
+        leaderboard[team.name] = sum(problems_solved_by_team)  # type: ignore[arg-type]
```

Comment:
Use `Team.all().annotate(team_points = Sum("solvedproblems__problem__points"))`
adds a team_points field in the Result

Reply:
![image](https://github.com/lugvitc/pwncore/assets/51919967/faae6a54-1d5a-4838-8819-21880f5bbb2e)
Working fine